docs/typescript/reference/client.md

```diff
@@ -0,0 +1,152 @@
+
+:::note 
+`DBOSClient` is included in the `@dbos-inc/dbos-sdk` package, the same package that used by DBOS applications.
+Where DBOS applications use the [static `DBOS` class](../reference/transactapi/dbos-class.md),
+external applications use the [`DBOSClient` class](../reference/client.md) instead.
+:::
+
+### class DBOSClient
+
+```ts
+interface EnqueueOptions {
+    workflowName: string;
+    workflowClassName: string;
+    queueName: string;
+    appVersion?: string;
+    workflowID?: string;
+}
+
+class DBOSClient {
+    static create(databaseUrl: string, systemDatabase?: string): Promise<DBOSClient>;
+    destroy(): Promise<void>;
+    async enqueue<T extends (...args: any[]) => Promise<any>>(
+        options: EnqueueOptions,
+        ...args: Parameters<T>
+    ): Promise<WorkflowHandle<Awaited<ReturnType<T>>>>
+    enqueue<T extends unknown[]>(options: EnqueueOptions, ...args: T): Promise<void>;
+    retrieveWorkflow<T = unknown>(workflowID: string): WorkflowHandle<Awaited<T>>;
+    send<T>(destinationID: string, message: T, topic?: string, idempotencyKey?: string): Promise<void>;
+    getEvent<T>(workflowID: string, key: string, timeoutSeconds?: number): Promise<T | null>;
+}
+```
+
+#### `create`
+
+You construct a `DBOSClient` with the static `create` function. 
+
+The `databaseUrl` parameter is a [standard PostgreSQL connection URI](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING-URIS)
+for the DBOS application database.
+DBOS Client also needs to connect to the [system database](../../explanations/system-tables.md) of your DBOS application.
+The system database is stored on the same database server as the application database and typically has the same name as your application database, but suffixed with `_dbos_sys`. 
+If you are using a non-standard system database name in your DBOS application, you must also provide the name to `DBOSClient.create`.
+
+Example: 
+
+```ts
+import { DBOSClient } from "@dbos-inc/dbos-sdk";
+
+const client = await DBOSClient.create("postgresql://postgres:password@localhost:5432/my_app_db");
+```
+
+#### `destroy`
+
+Asynchronously destroys a `DBOSClient` instance.
+
+#### `enqueue`
+
+Enqueues a workflow, similar to [passing a queue name to `startWorkflow`](./transactapi/dbos-class.md#starting-background-workflows)
+or using [`DBOS.withWorkflowQueue`](./transactapi/dbos-class.md#using-workflow-queues). 
+Like `startWorkflow`, the `enqueue` method returns a `WorkflowHandle` that you can use to retrieve the workflow results 
+asynchronously from an external application.
+
+When enqueuing a workflow from within a DBOS application, the workflow and queue metadata can be retrieved automatically.
+However, since `DBOSClient` runs outside the DBOS application, the metadata must be specified explicitly.
+This metadata includes:
+
+* `workflowName`: The name of the workflow method being enqueued.
```

Comment:
We should make it clearer that only `workflowName` and `queueName` are required, the rest are all optional.